pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -592,53 +585,37 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 
 	adsLog.Infof("Pushing %v", con.ConID)
 
-	_ = s.rateLimiter.Wait(context.TODO()) // rate limit the actual push
-
 	// Prevent 2 overlapping pushes.
 	con.pushMutex.Lock()
 	defer con.pushMutex.Unlock()
 
 	defer func() {
-		n := atomic.AddInt32(pushEv.pending, -1)
-		if n <= 0 && pushEv.push.End == timeZero {
-			// Display again the push status
-			pushEv.push.Mutex.Lock()
-			pushEv.push.End = time.Now()
-			pushEv.push.Mutex.Unlock()
-			proxiesConvergeDelay.Record(time.Since(pushEv.push.Start).Seconds())
-			out, _ := pushEv.push.JSON()
-			adsLog.Infof("Push finished: %v %s",
-				time.Since(pushEv.push.Start), string(out))
-		}
+		proxiesConvergeDelay.Record(time.Since(pushEv.push.Start).Seconds())
```

Comment:
What happens when push times out? 
We should dimension this metric with error-code. Otherwise if pushCDS errors out it will give rise to artificially low numbers.

Reply:
Good point

How would it look with error code? `proxy_converge{error=cds, time=5s}` and `proxy_converge{error=nil, time=5s}` (very rough example :slightly_smiling_face:)?

Do we even want to report this if there is an error?

Reply:
We want to report it, because there is some signal there.
We can always run query with status=ok, or status=cds_error